### PR TITLE
Generate SBOM manifest

### PR DESCRIPTION
**Pull request by RussKie**
Relates to https://github.com/dotnet/arcade/issues/8477

### CODE REVIEW

**Comment by epananth** [comment]
Can you please add a test build to this?

**Comment by RussKie** [comment]
https://dev.azure.com/dnceng/internal/_build/results?buildId=1631811&view=results

**Comment by epananth** [comment]
Test build did not generate sbom folders

**Comment by RussKie** [comment]
Here's a build with SBOM manifest: https://dev.azure.com/dnceng/internal/_build/results?buildId=1633615&view=logs&j=5ab303af-16db-5f58-82d4-945dcabe3bb5&t=24962100-60d4-5768-b736-1d1c025ebd15

@mmitche I added a SBOM step the  Prepare for Publish stage, but it's giving me a warning. Did I do it incorrectly?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -83,6 +83,12 @@ stages:
         pool:
           name: NetCore1ESPool-Internal
           demands: ImageOverride -equals Build.Server.Amd64.VS2019
+    - job: Generate_SBOM
```
Comment (epananth) [inline azure-pipelines.yml:86]: I don't think we need this here

**Comment by mmitche** [comment]
> Looked at the build
> 
> 1. SBOMs are getting generated in the Build stage
> 2. we don't need another sbom in prepare artifact stage ( I think all the artifacts are covered in previous stage)

Can we get rid of that stage then? If it's not doing anything interesting any longer. It used to do something around signing and packaging.

**Comment by epananth** [comment]
Yes will do

**Comment by epananth** [comment]
I was trying to remove the prepared artifacts stage, looks like there is some copying of intermediate packages in that job. I am not sure if that is required. I think it will be good to handle that stage removal in a separate PR

**Comment by RussKie** [comment]
Here's an updated build without SBOM in Prepare for Publish stage: https://dev.azure.com/dnceng/internal/_build/results?buildId=1637247&view=results

**Comment by RussKie** [comment]
In the interest of time I'm going to merge this, and we can follow up on anything outstanding later.

**Pull request merged**